Core/src/org/sleuthkit/autopsy/textextractors/TextFileExtractor.java

```diff
@@ -94,33 +102,59 @@ public boolean isSupported() {
-            // Tika first
+    /**
+     * Return the encoding of the file
+     * @return Detected encoding or UNKNOWN_CHARSET 
+     */
+    public Charset getEncoding() {
+        if (encoding != null)
+            return encoding;
+        
+        // Encoding detection is hard. We use several libraries since the data passed in is often messy.
+        
+        // First try CharsetDetector (from Tika / ICU4J)
+        // It is a rule-baesd detection approach
+        try (InputStream stream = new BufferedInputStream(new ReadContentInputStream(file))) {
             CharsetDetector detector = new CharsetDetector();
             detector.setText(stream);
             CharsetMatch tikaResult = detector.detect();
-            if (tikaResult != null && tikaResult.getConfidence() >= MIN_TIKA_MATCH_CONFIDENCE) {
+            if (tikaResult != null && tikaResult.getConfidence() >= MIN_CHARSETDETECT_MATCH_CONFIDENCE) {
                 try {
-                    return Charset.forName(tikaResult.getName());
-                } catch (UnsupportedCharsetException ignored) {
+                    encoding = Charset.forName(tikaResult.getName());
+                    return encoding;
+                } catch (UnsupportedCharsetException ex) {
+                    logger.log(Level.WARNING, "Error converting CharsetDetector Result", ex);
                 }
             }
+        } catch (IOException ignored) {
+            // IGNORE READ ERRORS HERE - Assume they were logged elsewhere
+        }
 
-            // Decodetect if Tika fails or falls below confidence threshold
+        // If that did not work, then use DecoDetect, which is stastical 
+        // We needed this for some Japanese text files that were incorrectly detected by CharsetDetector (with low confidence)
+        // This will not always work with messy data that combines some binary and some ASCII.
+        try {
             int maxBytes = 100000;
-            int numBytes = Math.min(stream.available(), maxBytes);
+            int numBytes = maxBytes;
+            if (file.getSize() < maxBytes) {
+                numBytes = (int) file.getSize();
+            }
+            
             byte[] targetArray = new byte[numBytes];
-            stream.read(targetArray);
+            file.read(targetArray, 0, numBytes);
             List<DecodetectResult> results = Decodetect.DECODETECT.getResults(targetArray);
             if (!results.isEmpty()) {
                 DecodetectResult topResult = results.get(0);
                 if (topResult.getConfidence() >= MIN_DECODETECT_MATCH_CONFIDENCE) {
-                    return topResult.getEncoding();
+                    encoding = topResult.getEncoding();
+                    return encoding;
                 }
             }
-        } catch (IOException ignored) {
+        } catch (TskCoreException ex) {
+            // IGNORE READ ERRORS HERE - Assume they were logged elsewhere
```

Comment:
@bcarrier, if there is an TskCoreException, it is thrown by AbstractFile and will not have been logged. I will fix this.